### DISCUSSION TITLE
Smart image classification

### DISCUSSION THREAD

**Discussion by quantified1995** [deckerst/aves#60] [Ideas]
Didn't want to add this to my previous post so I figured I'd post a separate (forgive me as I'm new to GitHub and app development)

Another incredible feature I would love to see in this app would be smart image classification. I've seen quite a few apps implement this and I consider it a make or break feature. I'm not sure to what extent the difficulty would be to implement this but it would be awesome to see.

**Comment by deckerst** [comment]
Welcome to GitHub! You're right to separate different requests in different issues/discussions. It's easier to manage.

By smart image classification, I guess you mean to automatically identify the kind of scenery and objects, like "beach", "car", "food", "logo"... If so, it's something I intend to do, likely with [ML Kit](https://developers.google.com/ml-kit/).

**Comment by chaptergy** [comment]
Would grouping of persons based on facial features be included in this feature idea? So being able to filter for photos with person A. Not sure whether the "identifier for each unique detected face" of ML Kit can be used for this, or if it is only meant for unique faces within a video sequence.

**Comment by deckerst** [comment]
@chaptergy I consider it a different feature, as does ML Kit:
- [face detection](https://developers.google.com/ml-kit/vision/face-detection)
- [image labeling](https://developers.google.com/ml-kit/vision/image-labeling)

That said, no need to create a new issue for it, as face detection by ML is something I'll look into when I get to #69. That ticket is for extracting faces already saved in the metadata, but to get a decent "people" feature, face detection by ML will be necessary from the start.
